guides/20240819_history_embedding.md

```diff
@@ -0,0 +1,237 @@
+- **Search History**: Perform searches using various filters and options.
+
+## Preparations
+
+### Installing Prerequisites
+
+Before you begin, ensure you have [Daytona](https://github.com/daytonaio/daytona) installed.
+
+You can install Daytona by running the following command in your terminal:
+
+```bash
+curl -L https://download.daytona.io/daytona/install.sh | sudo bash
+```
+
+Daytona creates a controlled [development environment](/definitions/d/development-environment) where you can work on the project with all dependencies correctly configured, reducing the likelihood of compatibility issues.
+
+### Exporting Browser History
+
+To work with your browser history, you'll need to export it from your browser:
+
+1. **Install the Export Chrome History Extension**: Download and install the [Export Chrome History](https://chrome.google.com/webstore/detail/export-chrome-history/) Chrome extension.
+2. **Export to CSV**: Use the extension to export your browser history as a CSV file. Save it to a convenient location, like `~/Downloads/history.csv`.
+
+Example CSV snippet:
+
+```csv
+visit_time,url,title,visit_count,typed_count
+2023-08-10 12:34:56,http://example.com,Example Site,5,2
+2023-08-09 11:22:33,http://anotherexample.com,Another Example,3,1
+```
+
+## Setting Up the Development Environment
+
+### Creating a Daytona Workspace
+
+With Daytona installed, you can create a [workspace](/definitions/d/daytona-workspace) for the [`nkkko/history` repository](https://github.com/nkkko/history):
+
+```bash
+daytona create https://github.com/nkkko/history --code
+```
+
+This command clones the [repository](/definitions/r/repository) and sets up the necessary environment. Daytona manages dependencies and configurations, ensuring that your environment is set up correctly for running the project scripts.
+
+### Set Up Environment Variables (Optional for Azure)
+
+If you plan to use Azure's [embedding](/definitions/d/embedding) services instead of the local model, you need to configure the environment variables. Create a `.env` file in the root directory of the [repository](/definitions/r/repository) and add the following:
+
+```env
+AZURE_API_VERSION=<your_azure_api_version>
+AZURE_ENDPOINT=<your_azure_endpoint>
+AZURE_OPENAI_API_KEY=<your_azure_api_key>
+```
+
+**Example**:
+```env
+AZURE_API_VERSION=2021-06-01-preview
+AZURE_ENDPOINT=https://your-resource-name.openai.azure.com/
+AZURE_OPENAI_API_KEY=your-secret-key
+```
+
+**Why Azure?**  
+Using Azure's [embedding](/definitions/d/embedding) service can be beneficial if you need higher performance or better semantic accuracy in your [embedding](/definitions/d/embedding)s, especially when dealing with large datasets.
+
+## Embedding and Searching Your Browser History
+
+### Embedding
+
+With your [development environment](/definitions/d/development-environment) ready, you can now embed your browser history into ChromaDB.
+
+To embed using the local model (all-MiniLM-L6-v2), run:
```

Comment:
In the meantime I have even changed the model as I have realised this one wasn't performing well for this use case. You can check my repo.